huntsman/observatory.py

```diff
@@ -472,43 +359,237 @@ def _create_autoguider(self):
 
         self.autoguider = guider
 
-    def _create_flat_field_observation(self, alt=None, az=None,
-                                       dither_pattern_offset=5 * u.arcmin,
-                                       dither_random_offset=0.5 * u.arcmin,
-                                       n_positions=9
-                                       ):
+    def _create_flat_field_observation(self, alt=None, az=None, *args, **kwargs):
+
+        # Specify coordinates for flat field
         if alt is None and az is None:
             flat_config = self.config['flat_field']['twilight']
             alt = flat_config['alt']
             az = flat_config['az']
+            self.logger.debug(f'Using flat-field alt/az from config.')
+        flat_coords = utils.altaz_to_radec(alt=alt, az=az, location=self.earth_location,
+                                           obstime=utils.current_time())
+        self.logger.debug(f'Making flat-field observation for alt/az: {alt:.03f}, {az:.03f}.')
+        self.logger.debug(f'Flat field coordinates: {flat_coords}')
+
+        # Create the Observation object
+        obs = DitheredFlatObservation(position=flat_coords, *args, **kwargs)
+        obs.seq_time = utils.current_time(flatten=True)
+        self.logger.debug("Flat-field observation: {}".format(obs))
+
+        return obs
+
+    def _take_autoflats(self, cameras, observation, safety_func,
+                        tolerance=0.1, target_scaling=0.17, bias=32,
+                        min_exptime=1*u.second, max_exptime=60*u.second,
+                        max_num_exposures=10, *args, **kwargs):
+        """Take flat fields iteratively by automatically estimating exposure times.
+
+        Args:
+            cameras (dict): Dict of camera name: Camera pairs.
+            filter_names (dict): Dict of filter name for each camera.
+            safety_func (func): Boolean function that returns True only if safe to continue.
+        """
+        # Get the target counts and tolerance for each camera
+        target_counts = {}
+        counts_tolerance = {}
+        for cam_name, cam in cameras.items():
+            try:
+                bit_depth = cam.bit_depth.to_value(u.bit)
+            except NotImplementedError:
+                self.logger.debug(f'No bit depth property for {cam_name}. Using 16.')
+                bit_depth = 16
+            target_counts[cam_name] = target_scaling * 2**bit_depth
+            counts_tolerance[cam_name] = tolerance * 2**bit_depth
+            self.logger.debug(f'Target counts for {cam_name}: '
+                              f'{target_counts[cam_name]}+-{counts_tolerance[cam_name]}.')
```

Comment:
Don't be afraid of using utf8. ;)

```suggestion
                              f'{target_counts[cam_name]}±{counts_tolerance[cam_name]}.')
```